sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ValueStateImpl.scala

```diff
@@ -52,14 +58,12 @@ class ValueStateImpl[S](
     val keyByteArr = toRow
       .apply(keyOption.get).asInstanceOf[UnsafeRow].getBytes()
 
-    val schemaForKeyRow: StructType = new StructType().add("key", BinaryType)
     val keyEncoder = UnsafeProjection.create(schemaForKeyRow)
```

Comment:
This seems to be missing spot on previous PR review - we need to do this only once instead of each method call. Please move to the field of class. Same for valueEncoder for encodeValue as well.

Reply:
OK to do in separate PR, as it's technically not in this scope of PR.

Reply:
Done